rfcs/0080-nixos-release-schedule

```diff
@@ -0,0 +1,69 @@
+---
+feature: nixos-release-schedule
+start-date: 2020-11-28
+author: Jonathan Ringer (@jonringer)
+co-authors: Frederik Rietdijk (@FRidh)
+related-issues:
+---
+
+# Summary
+[summary]: #summary
+
+NixOS is released twice a year; the current release schedule targets March
+and September for the release months. However, trying to provide a polished
+desktop-user experience is difficult and moving the target release months to
+May and November should minimize the amount of in-house work needed to
+cut a release.
+
+# Motivation
+[motivation]: #motivation
+
+By changing our release dates to follow those of certain upstream projects,
+we could reduce the amount of issues we encounter during stabilization.
+In particular, GNOME and KDE Plasma both have a release in September, which
+allows ample time for NixOS contributors to package and stabilize the
+desktop managers before the November release. For the May release, GNOME will
+have a similar stabilization period; while Plasma, which has a four month
+release cadence, will have a slightly older but still supported non-LTS
+release. Using LTS releases is not feasible as there is no guarantee that
+they will be compatible with the version of systemd during branch-off.
+
+The current March and September release months may be the worst months for
+GNOME and Plasma, as the release available during stabilization would be
+End-of-Life a few weeks after the target release date. Also, this period marks
+the greatest potential dependency "drift" from development on unstable. For example,
+many of the 20.09 release blockers were related to issues with systemd-246 and plasma-5.18,
+however, plasma-5.19 and the yet-released 5.20 both supported systemd-246 with
+zero additional effort.
+
+Choosing branch-off points in master where the major desktop manager use
+cases should be well stabilized will allow for release stabilization to
+focus on failing builds and minor issues.
+
+## Core changes
+
+The 21.03 release will be delayed until 21.05, and the current 20.09 release
+will be supported until June 2021. After the initial 21.05 release, all subsequent
+releases will occur six months apart following a YY.05 and YY.11 convention.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+There will be some initial confusion with stable users as to why the 21.03
+release was delayed.
+
+# Alternatives
+[alternatives]: #alternatives
+
+## Maintain the status quo regarding release dates
+
+We could keep the release dates in March and September. However, we
+would need some other way to minimize the work and risk from trying to
+provide a polished experience for 60,000+ packages, 17 desktop managers,
+and 30+ window managers.
+
+# Future work
+[future]: #future-work
+
```

Comment:
- Update [osinfo-db entry](https://gitlab.com/libosinfo/osinfo-db/-/blob/master/data/os/nixos.org/nixos-unstable.xml.in) with the changes

Reply:
Already captured in https://github.com/NixOS/nixpkgs/issues/97405 and outside of the scope of the RFC

Reply:
The issue only seems to deal with release process. I assume this RFC will mean changing https://github.com/NixOS/nixpkgs/blob/master/.version, which means we will need to update the osinfo-db entry for `nixos-unstable` to match.

Reply:
yes, this is mentioned here: https://nixos.org/manual/nixos/stable/index.html#at-beta-release-time with the heading "On the master branch, increment the .version file".

I will update the example command when transferring over to the https://github.com/NixOS/release-wiki

Reply:
Well, that is still only relevant to the 21.11 bump. Or will the unstable branch remain 21.03pre until then, even after this RFC has been accepted?

Reply:
>  Or will the unstable branch remain 21.03pre until then, even after this RFC has been accepted?

unstable branch will always be the pre suffix

only release branches will differ

Reply:
Yes. But it is not clear to me whether after accepting this RFC, the unstable branch will become `21.05pre` instead of `21.03pre` it currently is.

This is orthogonal to the beta release and will affect whether we have to update the osinfo-db.

Reply:
Yes, 21.03pre will need to be changed to 21.05pre. Sorry.

Reply:
added the related two work items. Sorry for misinterpreting.